packages/flutter/test/material/drawer_test.dart

```diff
@@ -969,6 +969,152 @@ void main() {
+      await tester.pumpWidget(
+        MaterialApp(
+          theme: ThemeData(platform: themePlatform),
+          home: Scaffold(
+            key: scaffoldKey,
+            drawer: const Drawer(child: Text('Drawer')),
+            body: Container(),
+          ),
+        ),
+      );
+      expect(find.text('Drawer'), findsNothing);
+
+      scaffoldKey.currentState!.openDrawer();
+      await tester.pumpAndSettle();
+      expect(find.text('Drawer'), findsOneWidget);
+    }
+
+    group('Semantics label', () {
+      Future<void> testDrawerSemanticsLabel({
+        required WidgetTester tester,
+        required TargetPlatform themePlatform,
+        required Matcher expected,
+      }) async {
+        const DefaultMaterialLocalizations localizations = DefaultMaterialLocalizations();
+
+        await pumpDrawer(tester: tester, themePlatform: themePlatform);
+
+        final Finder popupFinder = find.bySemanticsLabel(localizations.drawerLabel);
+        expect(popupFinder, expected);
+      }
+
+      testWidgets(
+        'Semantics label is null by default on Apple platforms',
+        (WidgetTester tester) async {
+          // When someone sets theme.platform to TargetPlatform.android on an Apple device,
+          // assistive technology like VoiceOver should work correctly by having
+          // a null semantics label value by default.
+          await testDrawerSemanticsLabel(
+            tester: tester,
+            themePlatform: TargetPlatform.android,
+            expected: findsNothing,
+          );
+        },
+        variant: const TargetPlatformVariant(<TargetPlatform>{
+          TargetPlatform.iOS,
+          TargetPlatform.macOS,
+        }),
+      );
+
+      testWidgets(
+        'Semantics label is non-null by default on non-Apple platforms',
+        (WidgetTester tester) async {
+          // When someone sets theme.platform to TargetPlatform.iOS on a non-Apple device,
+          // assistive technologies (Talk Back/NVDA/JAWS...) should work correctly
+          // by having a non-null semantics label value by default.
+          await testDrawerSemanticsLabel(
+            tester: tester,
+            themePlatform: TargetPlatform.iOS,
+            expected: findsOneWidget,
+          );
+        },
+        variant: const TargetPlatformVariant(<TargetPlatform>{
+          TargetPlatform.android,
+          TargetPlatform.fuchsia,
+          TargetPlatform.linux,
+          TargetPlatform.windows,
+        }),
+      );
+    });
+
+    group('Semantics barrier', () {
+      Future<void> testDrawerSemanticsBarrier({
+        required WidgetTester tester,
+        required TargetPlatform themePlatform,
+        required Matcher expected,
+      }) async {
+        final SemanticsTester semantics = SemanticsTester(tester);
+
+        await pumpDrawer(tester: tester, themePlatform: themePlatform);
+
+        expect(semantics, expected);
+
+        semantics.dispose();
+      }
+
+      testWidgets(
```

Comment:
you can combine this and the next by update the expect to be defaultTargetPlatform adaptive